### PR TITLE
Feature/interface

### PR DESCRIPTION
**Pull request by mr-ryan12**
What (if any) features are you implementing?
- Moved interfaces to their own file for reusability
- Attempted toggling dark mode functionality with the Navigation Bar. Thankfully works as intended

What (if anything) did you refactor?
- Refactored all component files to import interfaces as needed

Were there any issues that arose?
- Not at the time

Is there anything that you need from your teammate?
- PR review and comments

Any other comments, questions, or concerns?
- Will have to revisit further implementation of dark mode throughout the rest of the application

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -27,10 +26,24 @@ class App extends Component<{}, State> {
     }
   }
 
+  toggleDarkMode = () => {
```
Comment (StevieAmb) [inline src/App.tsx:29]: I love this function so much, and how it's being used to alter a classname. Wild.

File: src/Categories/Categories.tsx [hunk 2]
```diff
@@ -1,12 +1,9 @@
 import React from 'react'
 import CategoryCard from '../CategoryCard/CategoryCard'
+import { CategoriesProps } from '../interface'
```
Comment (StevieAmb) [inline src/Categories/Categories.tsx:3]: Brilliant use of the interface file. I also love that it helped you reuse the interfaces.

File: src/App.tsx (hunk 1 above)
Comment (Jayquintana) [inline src/App.tsx:29]: good stuff man !

File: src/Navigation/Navigation.scss [hunk 3]
```diff
@@ -1,30 +1,28 @@
...
-  padding: 20px;
-  display: flex;
-  width: 100%;
-  max-width: 66%;
-  margin: 0px;
+  @include behind-logo($primary-color, $secondary-color);
+}
+
+.behind-logo-dark {
+  @include behind-logo($primary-color-dark, $secondary-color-dark);
```
Comment (StevieAmb) [inline src/Navigation/Navigation.scss:17]: I love this use of Sass for the dark-mode of the page.

File: src/Navigation/Navigation.tsx [hunk 4]
```diff
@@ -1,15 +1,21 @@
...
   return (
-    <nav className="navigation-bar">
-      <div className="behind-logo">
+    <>
+    <nav className={"navigation-bar" + props.darkMode}>
+      <div className={"behind-logo" + props.darkMode}>
         <Link className='title-link' to='/' >
           <h1 className="logo">
-            <div className='art-from-the'>
+            <div className={"art-from-the" + props.darkMode}>
```
Comment (StevieAmb) [inline src/Navigation/Navigation.tsx:18]: Passing down a part of a classname and then concatenating it an already created classname is such a creative way to toggle.

**Review by StevieAmb — APPROVED** [review]

**Review by Jayquintana — APPROVED** [review]
looks great!

**Pull request merged**